tempodb/encoding/vparquet4/block_autocomplete.go

```diff
@@ -114,6 +128,128 @@ func createDistinctIterator(
 	return currentIter, nil
 }
 
+func createDistinctEventIterator(
+	makeIter makeIterFn,
+	tag traceql.Attribute,
+	primaryIter parquetquery.Iterator,
+	conditions []traceql.Condition,
+) (parquetquery.Iterator, error) {
+	var (
+		columnSelectAs    = map[string]string{}
+		iters             []parquetquery.Iterator
+		genericConditions []traceql.Condition
+	)
+
+	// TODO: Potentially problematic when wanted attribute is also part of a condition
+	//     e.g. { span.foo =~ ".*" && span.foo = }
+	addSelectAs := func(attr traceql.Attribute, columnPath, selectAs string) {
+		if attr == tag {
+			columnSelectAs[columnPath] = selectAs
+		} else {
+			columnSelectAs[columnPath] = "" // Don't select, just filter
+		}
+	}
+
+	for _, cond := range conditions {
+		// Intrinsic?
+		switch cond.Attribute.Intrinsic {
+		case traceql.IntrinsicEventName:
+			pred, err := createStringPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			iters = append(iters, makeIter(columnPathEventName, pred, columnSelectAs[columnPathEventName]))
+			addSelectAs(cond.Attribute, columnPathEventName, columnPathEventName)
```

Comment:
Should these 2 lines be flipped?  It looks like it reads `columnSelectAs` before setting it.    

The next thought is that maybe we don't need the `columnSelectAs` map at all.   Since this creates the iter directly in the case statement, we don't need to track it in the map.  `CreateDistinctSpan/Resource/Iterator` funcs do that because they create the iters later after assessing all conditions.  But `createDistinctTraceIterator` is doing it like here, and doesn't need the map.